analysis/analyze.py

```diff
@@ -868,7 +868,7 @@ def specific_leakage_test(random, callback, keys, LeaksOnly=True, mp=False):
             rdic_pos[c] = numpy.array(rdic_pos[c], dtype=numpy.int64)
 
         # Extract X in correct order
-        X = numpy.asarray([Xglob[k] for k in keys])
+        X = numpy.asarray([Xglob[k.get_bytes().decode().encode()] for k in keys])
```

Comment:
Just out of interest what error was raised without the fix?
And do you maybe have a minimal example to reproduce the issue?

Reply:
Yep, `get_bytes()` is necessary. That's a bug I introduced while rewriting parts in the analysis.

Then will keep for the the `.decode().encode()`. It shouldn't do any harm and maybe is able to catch the error you experienced.